src/util/qrdisplay.js

```diff
@@ -6,12 +6,14 @@ const getQRDataURI = (data) => {
   return 'data:image/png;charset=utf-8;base64, ' + pngBuffer.toString('base64')
 }
 
-const openQr = (data) => {
+const openQr = (data, stop) => {
```

Comment:
I believe it is necessary for any use case, and is not specific to our implementation (the last use was, but this change is for general use). If someone is creating a request (starts polling right away), they will probably provide some UI for canceling, by cancelling you need to be able to stop polling (or want you to)